orderer/common/multichannel/registrar.go

```diff
@@ -909,8 +943,10 @@ func (r *Registrar) RemoveChannel(channelID string) error {
 }
 
 func (r *Registrar) removeMember(channelID string, cs *ChainSupport) error {
-	err := r.ledgerFactory.Remove(channelID)
+	r.pendingRemoval[channelID] = true
+	err := r.ledgerFactory.Remove(channelID, r.removeChannelFromPendingRemovals)
 	if err != nil {
+		r.pendingRemoval[channelID] = false
```

Comment:
I think a go routine could be launched here, calling `r.ledgerFactory.Remove`, and when it is finished, it could    remove the channel ID from pendingRemoval, no?

Reply:
Yes most recent commits take this approach now.